components/hab/src/command/pkg/uninstall_impl.rs

```diff
@@ -330,12 +362,58 @@ fn maybe_delete<U>(ui: &mut U,
         }
         ExecutionStrategy::Run => {
             ui.status(Status::Deleting, &ident)?;
+            if uninstall_hook_mode != UninstallHookMode::Ignore {
+                run_uninstall_hook(ui, &install).await?;
+            }
             let pkg_dir = install.installed_path();
             do_clean_delete(&pkg_root_path, &pkg_dir)
         }
     }
 }
 
+async fn run_uninstall_hook<T>(ui: &mut T, package: &PackageInstall) -> Result<()>
```

Comment:
Maybe change the name to indicate that this *might* run the hook? Or factor out the number of versions check?